luarules/gadgets/unit_missile_counter.lua

```diff
@@ -16,57 +16,264 @@ if not gadgetHandler:IsSyncedCode() then
+	{
+		return (0);
+	}
+	signal SIG_OPEN;
+	opened = 1;
+	
+	//the rest of the opening script
+	
+	ready = 1;
+	set-signal-mask SIG_OPEN;
+	sleep 1000;
+	start-script Close();
+}
+
+----Every projectile in the volley will need to be a seperate weapon
+----Modify your AimWeapon script for all weapons that use this
+
+AimWeapon2(heading, pitch)
+{
+	var volleyCount;												//Only one AimWeapon in the volley group should have this section. This calls the script that calls the Lua and passes the required values
+	var weaponNum;
+	volleyCount = 6;												//Amount of weapons in the volley group (max burst size)
+	weaponNum = 2;													//Which weapon has the correct weapondef to pull data from. If you have multiple volley groups each one should pull from a different weapon so you can use it to seperate them in the shotCounter().
+	call-script lua_ShotCounter(volleyCount, weaponNum);
+	
+	Signal SIG_AIM_2;												//Seperate signal for each weapon
+	
+	if(opened == 0)													//Opens if it isn't already (use only if opening is required)
+	{
+		start-script Open(0);
+	}
+	
+	//weapon turning here
+	
+	while(canshoot > 0 AND NOT (weapon == 1))						//Waits for its turn and terminates loops once no more shots are available
+	{
+		sleep 1;
+	}
+	set-signal-mask SIG_AIM_2;										//Prevents a bug where it can shoot the first missile while still closed
+	
+	//Waiting for turn here
+	
+	if(canshoot == 0)												//Checks if shots are available
+	{
+		return (0);
+	}
+	if(ready == 0)													//Checks if fully open 	(use only if opening is required)
+	{
+		return (0);
+	}
+	return (1);
+}
+
+----Add this to FireWeapon script for all weapons that use this
+
+FireWeapon2()
+{
+	timeout = 1;
+	justFired = justFired + 1;										//Used to delay next opening of hatches, so they close between shots
+	canshoot = canshoot - 1;
+	start-script Timer();
+	start-script CloseTimer();
+	missilesFired = missilesFired + 1;
+	sleep (burstDelay - 80);
+	weapon = 2;
+}
+--]]
+
 local isCommander = {}
 
-local function missile_counter(unitID, unitDefID, team)
+local function shotCounter(unitID, unitDefID, team, volleyCount, weaponNum)
 	
 	local armored, armorMultiplier
-	local damage = Spring.GetUnitWeaponDamages (unitID, 2, Game.armorTypes.standard)
+	local damage = Spring.GetUnitWeaponDamages(unitID, weaponNum, Game.armorTypes.standard)
 	local health, maxHealth
 	local counter
-	local losstate
-	local works, isUserTarget, targetID = Spring.GetUnitWeaponTarget (unitID, 1) 
-	local burstrate = (Spring.GetUnitWeaponState (unitID, 2, "burstRate") * 1000)
+	local losState
+	local targetType, isUserTarget, targetID = Spring.GetUnitWeaponTarget(unitID, 1) 
+	local burstRate = (Spring.GetUnitWeaponState(unitID, weaponNum, "burstRate") * 1000)
 	local targetDefID
-	if type(targetID) == "number" then
-		losstate = Spring.GetUnitLosState(targetID, team, true)
+	if targetType == 1 then
```

Comment:
```suggestion
	if targetType == 1 then -- target is a unit
```
Add comment to explain cryptic value returned by the engine